op-guide/docker.md

```diff
@@ -1,9 +1,196 @@
 
-## Run with `docker-compose`
+## Preparation
 
-A simple `docker-compose.yml`:
+Before you start, make sure that:
+
++ Installed the latest version of [Docker](https://www.docker.com/products/docker) 
++ Pulled the TiDB/TiKV/PD docker images from PingCAP's Docker Hub repositories
+`docker pull pingcap/tidb:latest`
+`docker pull pingcap/tikv:latest`
+`docker pull pingcap/pd:latest`
+
+## Create a docker bridge network
+
+```bash
+net="isolated_nw"
+docker network rm ${net}
+docker network create --driver bridge ${net} 
+```
+
+After creating a docker network, we will make all the TiDB containers added to it, which composed a standalone cluster.
+The service in cluster can communicate with each other using the name instead of IP addresss.
+In addition, you can replace the network name above with any you like.
+
+## Using `Busybox` container as storage volume
+
+```bash
+docker run -d --name ti-storage \
+  -v /tidata \
+  busybox
+```
+
+## Start PD service
+
+First we start up 3 pd-servers, respectively named as **pd1**, **pd2**, **pd3**.
+
+**pd1:** 
+
+```bash
+docker run --net ${net} -d --name pd1 \
+  -v /usr/share/ca­certificates/:/etc/ssl/certs \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/pd \
+  --cluster-id=1 \
+  --name="pd1" \
+  --data-dir="/tidata/pd1" \
+  --client-urls="http://0.0.0.0:2379" \
+  --advertise-client-urls="http://pd1:2379" \
+  --peer-urls="http://0.0.0.0:2380" \
+  --advertise-peer-urls="http://pd1:2380" \
+  --initial-cluster="pd1=http://pd1:2380,pd2=http://pd2:2380,pd3=http://pd3:2380" \
+  --addr="0.0.0.0:1234" \
+  --advertise-addr="pd1:1234"
+```
+
+**pd2:**
+
+```bash
+docker run --net ${net} -d --name pd2 \
+  -v /usr/share/ca­certificates/:/etc/ssl/certs \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/pd \
+  --cluster-id=1 \
+  --name="pd2" \
+  --data-dir="/tidata/pd2" \
+  --client-urls="http://0.0.0.0:2379" \
+  --advertise-client-urls="http://pd2:2379" \
+  --peer-urls="http://0.0.0.0:2380" \
+  --advertise-peer-urls="http://pd2:2380" \
+  --initial-cluster="pd1=http://pd1:2380,pd2=http://pd2:2380,pd3=http://pd3:2380" \
+  --addr="0.0.0.0:1234" \
+  --advertise-addr="pd2:1234"
+```
+
+**pd3:**
+
+```bash
+docker run --net ${net} -d --name pd3 \
+  -v /usr/share/ca­certificates/:/etc/ssl/certs \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/pd \
+  --cluster-id=1 \
+  --name="pd3" \
+  --data-dir="/tidata/pd3" \
+  --client-urls="http://0.0.0.0:2379" \
+  --advertise-client-urls="http://pd3:2379" \
+  --peer-urls="http://0.0.0.0:2380" \
+  --advertise-peer-urls="http://pd3:2380" \
+  --initial-cluster="pd1=http://pd1:2380,pd2=http://pd2:2380,pd3=http://pd3:2380" \
+  --addr="0.0.0.0:1234" \
+  --advertise-addr="pd3:1234"
+```
+
+After that, if you need to add **pd4** into the existing cluster, just use `--join` flag, and specify any one of the available **advertise-client-urls** above.
```

Comment:
+2. If you need to add **pd4** into the existing cluster, use the `--join` flag, and specify any one of the available **advertise-client-urls** above.